src/trim_comparison.rs

```diff
@@ -0,0 +1,808 @@
+                node.append(rctree::Node::<usize>::new(usize::MAX));
+                let mut new_node = node.last_child().unwrap();
+                build_diff_map_inner(child, &mut new_node, diff_nodes);
+            }
+        }
+        Value::Object(map) => {
+            for (key, val) in map {
+                node.append(rctree::Node::<usize>::new(usize::MAX));
+                let mut new_node = node.last_child().unwrap().clone();
+                if key == DIFFERENT {
+                    let diff_node = node.last_child().unwrap().clone();
+                    diff_nodes.push(diff_node);
+                }
+                build_diff_map_inner(val, &mut new_node, diff_nodes);
+            }
+        }
+        _ => (),
+    }
+}
+
+/// Creates a tree mirroring the Value passed in but with nodes of usize.
+/// Each node value represents how far up a Value is from a Value that of type Different.
+/// Note that this is not the true distance from a Different, because distance is only updated up the tree and for siblings.
+/// For example, the following structure, where nodes are named by their "diff_distance":
+///       2
+///      / \
+///     1   1
+///    /     \
+///   ∞       Different (0)
+///  /        /   \
+/// ∞       base   native
+fn build_diff_map(value: &Value) -> rctree::Node<usize> {
+    let mut tree = rctree::Node::<usize>::new(usize::MAX);
+
+    // Track the Different nodes so we can propagate the distance from Different up.
+    let mut diff_nodes = vec![];
+    build_diff_map_inner(value, &mut tree, &mut diff_nodes);
+
+    for mut node in diff_nodes {
+        propagate_diff_up(&mut node, 0);
+    }
+
+    tree
+}
+
+/// Remove Same values from the tree based off of the diff_map and the config passed in.
+fn trim_value(value: Value, diff_map: &rctree::Node<usize>, config: &TrimConfig) -> Option<Value> {
+    match value {
+        Value::Object(map) => {
+            let trimmed_map = Map::<_, _>::from_iter(map.into_iter().enumerate().filter_map(
+                |(index, (key, child_val))| {
+                    let child_diff_map = diff_map.children().skip(index).next().unwrap();
+                    let trimmed_child = trim_value(child_val, &child_diff_map, config);
+                    trimmed_child.map(|trimmed_value| (key, trimmed_value))
+                },
+            ));
+            Some(serde_json::Value::Object(trimmed_map))
+        }
+        Value::Array(vec) => {
+            let trimmed_vec = vec
+                .into_iter()
+                .enumerate()
+                .filter_map(|(index, child_val)| {
+                    let child_diff_map = diff_map.children().skip(index).next().unwrap();
+                    trim_value(child_val, &child_diff_map, config)
+                })
+                .collect_vec();
+            Some(serde_json::Value::Array(trimmed_vec))
+        }
+        Value::String(val_str) if value_is_same(&value) => {
+            let dist_to_diff = *diff_map.borrow();
```

Comment:
Can you explain `*diff_map.borrow()`

Reply:
The rctree lib uses `Rc<RefCell>` internally. In order to access the value of a node, `rctree::Node` has a `borrow()`, which returns a `std::cell::Ref` containing the underlying value. Then the actual value can be accessed by dereferencing. It looks like this is the only interface provided for accessing values.